payment-in-stablecoins/README.md

```diff
@@ -0,0 +1,133 @@
+# Proposal Payment in Stablecoins
+
+## Simple Summary
+
+Allows the DAO to execute proposals that send funds denominated in USD. This change addresses the ETH/USD price volatility risk both sides suffer from today, as well as the DAO's ability to perform ETH-to-USD swaps trustlessly.
+
+## Abstract
+
+This specification introduces two new contracts that facilitate stablecoin payments:
+
+- `NOU` a new ERC20 token the DAO can mint to say "I owe you this amount of USD"
+- `NounsStablecoinPayments` a contract that:
+  - allows the DAO to mint NOU tokens as payment to proposal recipients
+  - allows anyone to sell it stablecoins in exchange for ETH with some additional incentive
+  - allows proposal recipients to redeem NOU for stablecoins
+
+## Technical Specification
+
+### Overview
+
+#### Interaction flow
+
+1. A proposal is created, where the DAO wants to pay the builder $100K:
+   1. Assuming ETH/USD is $1500, and a volatility buffer of 2x
+   2. The proposal transaction would be:
+      `NounsStablecoinPayments.mintIOUs(recipient, 100_000)`
+      where `msg.value = NounsStablecoinPayments.ethNeeded(100_000, 2)`
+2. Upon proposal execution:
+   1. Proposal recipient is minted 100K NOU tokens
+   2. 133.33 ETH is sent from the treasury to NounsStablecoinPayments
+3. Arbitrageurs sell USD to NounsStablecoinPayments in exchange for ETH:
+   1. They call `NounsStablecoinPayments.sellStablecoin(100_000)` (or smaller amounts across multiple txs)
+      1. Expected to happen when `NounsStablecoinPayments.price()` is better than other DEX prices
+   2. NounsStablecoinPayments's USD balance grows by $100K, and an appropriate ETH balance is spent
+4. Proposal recipient redeems USD in exchange for NOU tokens:
+   1. They call `NounsStablecoinPayments.redeem()`
+   2. 100K NOU tokens are burned, and 100K USD is transferred from NounsStablecoinPayments to recipient
+
+#### `NOU` contract
+
+A new ERC20 token contract:
+
+- Inherits from OpenZeppelin's `AccessControlEnumerable`
+  - Assigns the `ADMIN_ROLE` to an `admin` provided in the constructor; admin can administer its own role and all other roles
+- `function mint(address recipient, uint amount)`
+  - Mints `amount` tokens to `recipient`
+  - Permits only senders of role `MINTER_ROLE`
+- `function burn(address from, uint amount)`
+  - Burns `amount` tokens from `from`'s account
+  - Permits only senders of role `BURNER_ROLE`
+
+#### `NounsStablecoinPayments` contract
+
+A new contract that helps the DAO acquire stablecoins by buying them from bots for a margin, and helps proposal recipients redeem stablecoin payments.
+
+##### Contract state variables
+
+Immutable:
+
+- `iouToken` the address of the NOU token ERC20 contract
+- `stablecoin` the address of the ERC20 token to use as USD, e.g. USDC or DAI
```

Comment:
We'll need the stablecoin decimals for the price calculation, right?

Reply:
good point. clarifying, you're saying it's not a good idea to call `token.decimals()` because:

1. to save on gas
2. some tokens might not implement this function 

is that right?

Reply:
Yeah, basically no point to make this an external call because it's static.

Reply:
great, added

Reply:
I think soli meant the save the decimals of the stablecoin, e.g DAI

Reply:
Yes, that's right. Referring to the stablecoin.

Reply:
of course, I don't know why I got confused there :)